storybookjs/storybook
# Perf: Replace react-hotkeys with useEffect keybinding

Issue: -

## What I did

This gets rid of react-hotkeys, a 310kb dependency. It was used only once, for a global Escape key listener. That's not even react-hotkeys' intended use case, because it's mainly useful for local (focus-based) events.

I replaced it with a simple `document.addEventListener` wrapped in `useEffect`, just like the ones in the sidebar.

## How to test

Go to one of the settings pages (e.g. "About your Storybook") and press the Escape key. The page should close and you should return to the main preview UI.

- Is this testable with Jest or Chromatic screenshots? no
- Does this need a new example in the kitchen sink apps? no
- Does this need an update to the documentation? no

If your answer is yes to any of these, please make sure to include it in your PR.
